### PR TITLE
CLOUDP-70872: Adding Third-Party Integration Settings

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Implemented [Third-Party Integration Settings](https://docs.atlas.mongodb.com/reference/api/third-party-integration-settings/) 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/third_party_integration.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+)
+
+const (
+	integrationBasePath = "groups/%s/integrations"
+)
+
+// IntegrationsService is an interface for interfacing with the Third-Party Integrations
+// endpoints of the MongoDB Atlas API.
+
+// See more: https://docs.atlas.mongodb.com/reference/api/third-party-integration-settings/
```
Comment (gssbzn) [inline mongodbatlas/third_party_integration.go:16]:
```suggestion
// IntegrationsService provides access to the third party integrations related functions in the Atlas API. 
//
// See more: https://docs.atlas.mongodb.com/reference/api/third-party-integration-settings/
```

File: mongodbatlas/third_party_integration.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+type IntegrationsService interface {
+	Create(context.Context, string, string, *ThirdPartyService) (*IntegrationResponse, *Response, error)
+	Replace(context.Context, string, string, *ThirdPartyService) (*IntegrationResponse, *Response, error)
+	Delete(context.Context, string, string) (*Response, error)
+	Get(context.Context, string, string) (*ThirdPartyService, *Response, error)
+	List(context.Context, string) (*IntegrationResponse, *Response, error)
+}
+
+// TeamsServiceOp handles communication with the Teams related methods of the
+// MongoDB Atlas API
```
Comment (gssbzn) [inline mongodbatlas/third_party_integration.go:26]:
```suggestion
// IntegrationsServiceOp provides an implementation of the IntegrationsService interface
```

**Review by colm-quinn — APPROVED** [review]
LGTM, though I don't have a lot of context on the API you're using here.

File: mongodbatlas/third_party_integration.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+type IntegrationsServiceOp service
+
+var _ IntegrationsService = &IntegrationsServiceOp{}
+
+// IntegrationRequest contains parameters for different third-party services
```
Comment (gssbzn) [inline mongodbatlas/third_party_integration.go:31]:
```suggestion
// ThirdPartyService contains parameters for different third-party services
```

File: mongodbatlas/third_party_integration.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+	ChannelName string `json:"channelName,omitempty"`
+	RoutingKey  string `json:"routingKey,omitempty"`
+	FlowName    string `json:"flowName,omitempty"`
+	OrgName     string `json:"orgName,omitempty"`
+	URL         string `json:"url,omitempty"`
+	Secret      string `json:"secret,omitempty"`
+}
+
+// IntegrationResponse contains the response from the endpoint
+type IntegrationResponse struct {
```
Comment (gssbzn) [inline mongodbatlas/third_party_integration.go:52]: prefer plural for collections
```suggestion
type IntegrationsResponse struct {
```

File: mongodbatlas/third_party_integration.go [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+	Links      []*Link              `json:"links"`
+	Results    []*ThirdPartyService `json:"results"`
```
Comment (gssbzn) [inline mongodbatlas/third_party_integration.go:54]: [nit] the wrapper is `Integration`, and the collection `ThirdPartyService`
can you try to align them to a common term, ie: `IntegrationsResponse` and `Integration` or
`ThirdPartyIntegrations` and `ThirdPartyIntegration` (plural vs singular) or something like that

**Review by gssbzn — CHANGES_REQUESTED** [review]
just a couple of small nits

**Review by gssbzn — APPROVED** [review]
LGTM, @themantissa

**Review by themantissa — APPROVED** [review]
LGTM, We'll be using this for Terraform too pretty so let's have one of the devs give it a quick look.  Will add

**Review by leofigy — APPROVED** [review]
LGTM :)

**Pull request merged**